src/main/java/com/google/devtools/build/lib/analysis/config/transitions/TransitionFactory.java

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.devtools.build.lib.analysis.config.transitions;
+
+import com.google.devtools.build.lib.analysis.config.transitions.TransitionFactory.TransitionFactoryData;
+
+/**
+ * Helper for the types of transitions that are statically declared but must be instantiated for
```

Comment:
Since, as I understand, we're going to funnel *all* transitions through this interface, even if they can be instantiated statically, can we provide more context for this javadoc?

Also, let's reference https://github.com/bazelbuild/bazel/blob/1f684e1b87cd8881a0a4b33e86ba66743e32d674/src/main/java/com/google/devtools/build/lib/analysis/config/transitions/PatchTransition.java#L31-L33 (the importance of resolving to singletons when possible, otherwise being careful to support .equals), lest we run into hard-to-track bugs when these get passed around hash maps, etc.

Reply:
Added more details and added that note.

I'm still not entirely sure if everything will be factories, or if we'll still have static transitions. I can come back and update this Javadoc once a decision becomes clear during the implementation.